Various library link fixes

ZeroAbility opened the pull request:
Various fixes to the CMake project files, and some sources with incorrect values or paths.

Commit:
src/murmur: add missing libraries, components and definitions for build with vcpkg

davidebeatrici commented on src/mumble/CMakeLists.txt:
```diff
@@ -399,32 +399,39 @@ if(bundled-opus)
+		set_target_properties(opus PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+
+		target_link_libraries(mumble PRIVATE opus)
+	elseif(MSVC)
```
This is fine, but I would use `elseif(WIN32)` instead.

We will probably add support for MinGW in future.

Alternatively:
```cmake
	add_dependencies(mumble opus)
	target_include_directories(mumble PRIVATE "${3RDPARTY_DIR}/opus-src/include")

	if(WIN32)
		set_target_properties(opus PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
	else()
		set_target_properties(opus PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
	endif()
```

ZeroAbility replied:
Addressed with latest push

Commit:
src/mumble: replace target_link_libraries() for Opus and CELT

target_link_libraries() automatically adds the library's include directories to the specified target and marks it as dependency.

Both steps are fine, however there's a third we don't need: linking to the library, because we load Opus and CELT at runtime.

To make it clear that we don't link to the libraries, this commit replaces target_link_libraries() with add_dependencies() and target_include_directories().

Commit:
src/mumble_exe: don't use find_library() for Windows libraries

It's not required because the linker finds the libraries into the SDK's directory.

This is also done for consistency with the overlay project. We can't use find_library() in that as it would break 32 bit cross-compilation (due to it finding 64 bit libraries).
